src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -2698,6 +2754,393 @@ private void UpdateDatabase(ClusterOperationContext context, string type, Blitta
+            {
+                long total = 0;
+
+                using (Slice.From(context.Allocator, "db/", out var loweredPrefix))
+                {
+                    foreach (var result in items.SeekByPrimaryKeyPrefix(loweredPrefix, Slices.Empty, 0))
+                    {
+                        var (_, _, record) = GetCurrentItem(context, result.Value);
+
+                        switch (resultType)
+                        {
+                            case TotalType.StaticIndex:
+                                if (record.TryGet(nameof(DatabaseRecord.Indexes), out BlittableJsonReaderObject obj) && obj != null)
+                                    total += obj.Count;
+                                break;
+                            case TotalType.AutoIndex:
+                                if (record.TryGet(nameof(DatabaseRecord.AutoIndexes), out obj) && obj != null)
+                                    total += obj.Count;
+                                break;
+                            case TotalType.CustomSorters:
+                                if (record.TryGet(nameof(DatabaseRecord.Sorters), out obj) && obj != null)
+                                    total += obj.Count;
+                                break;
+                            case TotalType.Analyzers:
+                                if (record.TryGet(nameof(DatabaseRecord.Analyzers), out obj) && obj != null)
+                                    total += obj.Count;
+                                break;
+                            default:
+                                throw new ArgumentOutOfRangeException(nameof(type), type, null);
+                        }
+                    }
+
+                    return total;
+                }
+            }
+        }
+
+        private bool CanAssertLicenseLimits(ClusterOperationContext context, int minBuildVersion)
+        {
+            var licenseLimitsBlittable = Read(context, ServerStore.LicenseLimitsStorageKey);
+            if (licenseLimitsBlittable == null)
+                return false;
+
+            var licenseLimits = JsonDeserializationServer.LicenseLimits(licenseLimitsBlittable);
+            if (licenseLimits.NodeLicenseDetails == null)
+                return false;
+
+            var clusterTopology = _parent.GetTopology(context);
+
+            foreach (var clusterNode in clusterTopology.Members.Union(clusterTopology.Watchers).Union(clusterTopology.Promotables))
+            {
+                if (licenseLimits.NodeLicenseDetails.ContainsKey(clusterNode.Key) == false)
+                    return false;
+            }
+
+            foreach (var limit in licenseLimits.NodeLicenseDetails)
+            {
+                if (limit.Value.BuildInfo == null)
+                    return false;
+
+                if (limit.Value.BuildInfo.BuildVersion < minBuildVersion)
+                    return false;
+            }
+
+            return true;
+        }
+
+        private enum TotalType
```

Comment:
we need a better name, total type of what?

Reply:
done